### PR TITLE
OCM-8914 | test: Automated id:41549,52352,52764,66750

### PR DESCRIPTION
**Pull request by xueli181114**

### CODE REVIEW

**Comment by xueli181114** [comment]
`lixue@Xue-Lis-MacBook-Pro rosa % ginkgo -focus "(52764|41549|52352)" tests/e2e 
Running Suite: ROSA CLI e2e tests suite - /Users/lixue/Workspace/rosa/tests/e2e
===============================================================================
Random Seed: 1719218421

Will run 3 of 116 specs
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSS•SSSSSSSSS•time="2024-06-24 16:42:47" level=info msg="Trying to list subnets of the vpc"
time="2024-06-24 16:42:47" level=info msg="Got 6 subnets"
time="2024-06-24 16:42:47" level=info msg="subnet-0892e6de6c0353b8d\t10.0.2.0/24\tus-west-2b\t"
time="2024-06-24 16:42:47" level=info msg="subnet-02b172add189471f8\t10.0.0.0/24\tus-west-2a\t"
time="2024-06-24 16:42:47" level=info msg="subnet-0e383419a439d6086\t10.0.4.0/24\tus-west-2c\t"
time="2024-06-24 16:42:47" level=info msg="subnet-09367649dc78fd6e2\t10.0.1.0/24\tus-west-2a\t"
time="2024-06-24 16:42:47" level=info msg="subnet-07621515cf444bed3\t10.0.5.0/24\tus-west-2c\t"
time="2024-06-24 16:42:47" level=info msg="subnet-0343c59b24801abcf\t10.0.3.0/24\tus-west-2b\t"
time="2024-06-24 16:42:47" level=info msg="Got custom rt rtb-0d41d714714d54fd8 "
time="2024-06-24 16:42:47" level=info msg="Got custom rt rtb-00baa27ef2e2fcfde "
time="2024-06-24 16:42:47" level=info msg="Got custom rt rtb-0af423129a491fc70 "
time="2024-06-24 16:42:47" level=info msg="Got custom rt rtb-0e0129f1072523be3 "
time="2024-06-24 16:42:47" level=info msg="Got custom rt rtb-09d1e7522a794697c "
time="2024-06-24 16:42:47" level=info msg="Got custom rt rtb-0ccde1b483e29aa0f "
time="2024-06-24 16:42:47" level=info msg="Got main association for rt rtb-036ebd64b784bad1e"
time="2024-06-24 16:42:48" level=info msg="Going to prepare proper pair of subnets"
time="2024-06-24 16:42:48" level=info msg="Subnet subnet-0892e6de6c0353b8d in zone: us-west-2b, region us-west-2"
time="2024-06-24 16:42:48" level=info msg="Subnet subnet-02b172add189471f8 in zone: us-west-2a, region us-west-2"
time="2024-06-24 16:42:48" level=info msg="Got public subnet subnet-02b172add189471f8 and set it to the result"
time="2024-06-24 16:42:48" level=info msg="Subnet subnet-0e383419a439d6086 in zone: us-west-2c, region us-west-2"
time="2024-06-24 16:42:48" level=info msg="Subnet subnet-09367649dc78fd6e2 in zone: us-west-2a, region us-west-2"
time="2024-06-24 16:42:48" level=info msg="Got private subnet subnet-09367649dc78fd6e2 and set it to the result"
time="2024-06-24 16:42:48" level=info msg="Subnet subnet-07621515cf444bed3 in zone: us-west-2c, region us-west-2"
time="2024-06-24 16:42:48" level=info msg="Subnet subnet-0343c59b24801abcf in zone: us-west-2b, region us-west-2"
time="2024-06-24 16:42:53" level=info msg="Tag resource subnet-09367649dc78fd6e2 successfully"
•SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS

Ran 3 of 116 Specs in 172.777 seconds
SUCCESS! -- 3 Passed | 0 Failed | 0 Pending | 113 Skipped
PASS

Ginkgo ran 1 suite in 3m5.240921147s
Test Suite Passed`

**Comment by xueli181114** [comment]
```
lixue@Xue-Lis-MacBook-Pro rosa % ginkgo -focus 66750 tests/e2e
Running Suite: ROSA CLI e2e tests suite - /Users/lixue/Workspace/rosa/tests/e2e
===============================================================================
Random Seed: 1719218699

Will run 1 of 116 specs
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS•SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS

Ran 1 of 116 Specs in 78.316 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 115 Skipped
PASS

Ginkgo ran 1 suite in 1m24.441488133s
Test Suite Passed
```

File: tests/e2e/test_rosacli_machine_pool.go [hunk 1]
```diff
@@ -253,6 +413,52 @@ var _ = Describe("Edit machinepool",
...
+						maxNum, err := strconv.Atoi(max)
+						Expect(err).ToNot(HaveOccurred())
+						minReplicas += minNum
+						maxReplicas += maxNum
+						continue
+					}
+					replicasNum, err := strconv.Atoi(mp.Replicas)
+					Expect(err).ToNot(HaveOccurred())
+					minReplicas += replicasNum
+					maxReplicas += maxReplicas
```
Comment (radtriste) [inline tests/e2e/test_rosacli_machine_pool.go:440]:
```suggestion
					maxReplicas += replicasNum
```
?

File: tests/e2e/test_rosacli_machine_pool.go [hunk 2]
```diff
@@ -253,6 +413,52 @@ var _ = Describe("Edit machinepool",
 			clusterID          string
 			rosaClient         *rosacli.Client
 			machinePoolService rosacli.MachinePoolService
+
+			verifyClusterComputeNodesMatched func(clusterID string) = func(clusterID string) {
```
[the remark this answers is not in the document]
Comment (xueli181114) [reply tests/e2e/test_rosacli_machine_pool.go:417]: @radtriste This is clarified in `var` thus `:=` will not work.

File: tests/e2e/test_rosacli_machine_pool.go [hunk 3]
```diff
@@ -253,6 +413,52 @@ var _ = Describe("Edit machinepool",
...
+					if mp.AutoScaling == "true" {
+						autoscaling = true
+						min, max := strings.Split(mp.Replicas, "-")[0], strings.Split(mp.Replicas, "-")[1]
+						minNum, err := strconv.Atoi(min)
+						Expect(err).ToNot(HaveOccurred())
+						maxNum, err := strconv.Atoi(max)
+						Expect(err).ToNot(HaveOccurred())
+						minReplicas += minNum
+						maxReplicas += maxNum
+						continue
```
[the remark this answers is not in the document]
Comment (xueli181114) [reply tests/e2e/test_rosacli_machine_pool.go:435]: @radtriste I didn't see differences or risks here. Can you elaborate more?

File: tests/e2e/test_rosacli_machine_pool.go (hunk 1 above)
Comment (xueli181114) [reply tests/e2e/test_rosacli_machine_pool.go:440]: Aha, good catch.

File: tests/e2e/test_rosacli_machine_pool.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (radtriste) [reply tests/e2e/test_rosacli_machine_pool.go:435]: More readable IMO.
You code is typically doing an `if...else`, I am not expecting a continue in there to perform the same behavior

File: tests/e2e/test_rosacli_machine_pool.go (hunk 1 above)
Comment (xueli181114) [reply tests/e2e/test_rosacli_machine_pool.go]: Fixed

File: tests/e2e/test_rosacli_machine_pool.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (xueli181114) [reply tests/e2e/test_rosacli_machine_pool.go]: Updated.

**Review by radtriste — APPROVED** [review]
lgtm

**Pull request merged**